packages/devtools_app/test/table_test.dart

```diff
@@ -179,6 +184,114 @@ void main() {
+        keyFactory: (data) => Key(data.name),
+        sortColumn: flatNameColumn,
+        sortDirection: SortDirection.ascending,
+      );
+      await tester.pumpWidget(wrap(
+        SizedBox(
+          width: 800.0,
+          height: 200.0,
+          child: table,
+        ),
+      ));
+      expect(find.byWidget(table), findsOneWidget);
+      {
+        final FlatTableState<TestData> state =
+            tester.state(find.byWidget(table));
+        final columnWidths = state.computeColumnWidths(800.0);
+        expect(columnWidths.length, equals(3));
+        expect(columnWidths[0], equals(300.0));
+        expect(columnWidths[1], equals(400.0));
+        expect(columnWidths[2], equals(36.0));
+      }
+
+      // TODO(jacobr): add a golden image test.
+
+      await tester.pumpWidget(wrap(
+        SizedBox(
+          width: 200.0,
+          height: 200.0,
+          child: table,
+        ),
+      ));
+
+      {
+        final FlatTableState<TestData> state =
+            tester.state(find.byWidget(table));
+        final columnWidths = state.computeColumnWidths(200.0);
+        expect(columnWidths.length, equals(3));
+        expect(columnWidths[0], equals(300.0)); // Fixed width column.
+        expect(columnWidths[1], equals(400.0)); // Fixed width column.
+        expect(columnWidths[2], equals(0.0)); // Variable width column.
+      }
+
+      // TODO(jacobr): add a golden image test.
+    });
+
+    testWidgets('displays with multiple wide columns',
+        (WidgetTester tester) async {
+      final table = FlatTable<TestData>(
+        columns: [
+          flatNameColumn,
+          _WideMinWidthColumn(),
+          _NumberColumn(),
+          _WideColumn(),
+        ],
+        data: flatData,
+        onItemSelected: noop,
+        keyFactory: (data) => Key(data.name),
+        sortColumn: flatNameColumn,
+        sortDirection: SortDirection.ascending,
+      );
+      await tester.pumpWidget(wrap(
+        SizedBox(
+          width: 1000.0,
+          height: 200.0,
+          child: table,
+        ),
+      ));
+      expect(find.byWidget(table), findsOneWidget);
+      {
+        final FlatTableState<TestData> state =
+            tester.state(find.byWidget(table));
+        final columnWidths = state.computeColumnWidths(1000.0);
+        expect(columnWidths.length, equals(4));
+        expect(columnWidths[0], equals(300.0)); // Fixed width column.
+        expect(columnWidths[1], equals(160.0)); // Min width wide column
+        expect(columnWidths[2], equals(400.0)); // Fixed width column.
+        expect(columnWidths[3], equals(60.0)); // Variable width wide column.
+      }
+
+      await tester.pumpWidget(wrap(
+        SizedBox(
+          width: 200.0,
+          height: 200.0,
+          child: table,
+        ),
+      ));
+      {
+        final FlatTableState<TestData> state =
+            tester.state(find.byWidget(table));
+        final columnWidths = state.computeColumnWidths(200.0);
```

Comment:
why isn't the max width of 200.0 respected here? the sum of the below column widths are greater than 200?

Reply:
the max width is not respected if you have fixed width columns that sum up to more than the available width. @terrylucas this is one motivation for supporting fractional width columns as well for cases where we have opinions on the relative widths of columns rather than just fixed and min widths.